[fix bug 1173432] Workable intergration for MoFo.

glogiotatidis opened the pull request:
- [x]  tests

glogiotatidis commented:
@pmclanahan r?

pmclanahan commented:
Wow. This is quite a chunk of code for a codebase I don't know. I don't see any obvious issues. I assume it works well for you? The tests look good. Is there something I should do to test locally? We should probably setup Travis to run PR tests.

glogiotatidis commented:
aaand with travis ci!

glogiotatidis commented:
aaand with travis ci!

 I have this demo account in workable 
```
WORKABLE_URI = 'https://www.workable.com/spi/v2/accounts/mozilla-demo-1/jobs/'
WORKABLE_API_KEY = '504be02982ee74e69d7bb08b96c7e83b3f900eea9e8cda04bb6620e80c5be274'
```

If you want to test it locally, use these settings and run `./manage.py syncworkable` and optionally `./manage.py syncjobvite` to get the jobs from jobvite as well. All positions from both workable and jobvite should be listed in `our current listings` locally.

pmclanahan commented on careers/careers/helpers.py:
```diff
@@ -0,0 +1,15 @@
+
+
+@register.filter
+def position_get_absolute_url(obj):
```
Why not define the `get_absolute_url()` method for the workable model?

pmclanahan commented on careers/careers/views.py:
```diff
@@ -12,22 +15,37 @@ def home(request):
     return render(request, 'careers/listings.html', {
-        'positions': Position.objects.order_by('category__name', 'title'),
+        'positions': utils.get_all_positions(
+            order_by=lambda x: '{} {}'.format(x.category.name, x.title)),
```
is careers on python 2.7?

glogiotatidis replied:
nice catch

pmclanahan commented on careers/careers/views.py:
```diff
@@ -1,8 +1,11 @@
 
-from django_jobvite.models import Position
+from django_jobvite import models as jobvite_models
+from careers.django_workable import models as workable_models
```
NIT: should be organized in the next section with the other careers imports.

pmclanahan commented on careers/careers/views.py:
```diff
@@ -12,22 +15,37 @@ def home(request):
-        })
-    except Position.DoesNotExist:
-        raise Http404
+    # Cannot use __exact instead of __contains due to MySQL collation
```
WAT?!

glogiotatidis replied:
Ha yes, the joys of mysql. I updated to comment with more information. I was trying to understand why edna had this as `__contains` instead of `__exact` and the reason is that the way mysql is configured for careers does not allow case sensitive matching. Using `__contains` works around that limitation.

pmclanahan replied:
:arrow_up: shakes fist at mysql :facepunch:

pmclanahan commented:
Looks good to me. I just had that one little thing about not needing the helper filter since you could make the models act the same with a `get_absolute_url()` method, but since you've already done the work, it's just a NIT.

r+

glogiotatidis commented:
> Looks good to me. I just had that one little thing about not needing the helper filter since you could make the models act the same with a get_absolute_url() method, but since you've already done the work, it's just a NIT.

I was debating whether this should be a standalone django app, as django_jobvite, and then probably some how I ended up with this solution which doesn't make sense anyway. I removed the helper, added `get_absolute_url` and things work just fine. 

Thanks for the review! :)
